### DISCUSSION TITLE
How to set custom text for login button

### DISCUSSION THREAD

**Discussion by aaaguirrep** [oidc-wp/openid-connect-generic#306] [General]
Hi, is there any options to set the button text "Login with OpendID Connect" for custom text? For instance: "Login with Corporate SSO" u other text.

Thanks in advance.

Regards,
Arsenio

**Comment by timnolte** [comment]
Yes, this is documented in the [HowTo](https://www.github.com/oidc-wp/openid-connect-generic/tree/dev/HOWTO.md) as well as the [Wiki](https://github.com/oidc-wp/openid-connect-generic/wiki/Hooks---Actions-&-Filters#openid-connect-generic-login-button-text).

**Reply by aaaguirrep** [reply]
Thanks for the information, sorry but I am a newbie on wordpress, do i need to modify a specific plugin's file or is there a specific process to add the filter and dont touch the plugin's file?

Thanks for your help.

Regards,
Arsenio

**Reply by timnolte** [reply]
So I have a customization Must Use plugin that has this sort of functionality that uses the hooks we provide. https://github.com/timnolte/oidc-keycloak-sso/blob/a8fd2761eaaea5968ecdb878c1939d57a6d199f3/oidc-keycloak-custom.php#L16

I'll put together a version of that add-on that will provide just the button text settings customization which you should be able to drop in and use on your site.

**Reply by aaaguirrep** [reply]
It will be a great feature, setup a custom text on OpenID Connect identity provider settings.

**Reply by timnolte** [reply]
@aaaguirrep so you can get an add-on [here](https://github.com/oidc-wp/oidc-generic-button-text-addon/tree/main) which will add an additional plugin setting to configure the button text. The README file there has the installation instructions.

**Reply by AdamTheRedBaron** [reply]
The other option is just to create a button that uses the plain url shortcode as the link target. Then you can style any way that you want to.

**Reply by timnolte** [reply]
I assumed the original question was related to the SSO Login button that shows up on the WordPress login screen. In that case changing the text requires the use of the hook that is available. Also, the MU Plugin I provided is basically just leveraging the hooks we have already made available but not everyone has the skill set or experience to use those.

**Reply by AdamTheRedBaron** [reply]
@timnolte you're probably right! I'm being blinkered by my own use-case scenario.

**Comment by aaaguirrep** [comment]
Thanks for your comments I have modified the text by adding the filter on functions.pho file.
________________________________
